ec/ecdatasource/deploymentdatasource/schema_apm.go

```diff
@@ -18,72 +18,116 @@
 package deploymentdatasource
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func newApmResourceInfo() *schema.Resource {
-	return &schema.Resource{
-		Schema: map[string]*schema.Schema{
+func apmResourceInfoSchema() tfsdk.Attribute {
+	return tfsdk.Attribute{
+		Description: "Instance configuration of the APM type.",
+		Computed:    true,
+		Validators:  []tfsdk.AttributeValidator{listvalidator.SizeAtMost(1)},
+		Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
 			"elasticsearch_cluster_ref_id": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "The user-specified ID of the Elasticsearch cluster to which this resource kind will link.",
```

Comment:
Can we update any references of `resource kind` to `APM resource`

Reply:
@pascal-hofmann ? I fixed this description, but there are still other similar places in the file.